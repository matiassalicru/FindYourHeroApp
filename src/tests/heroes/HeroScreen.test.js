import React from "react";
import { MemoryRouter, Route } from "react-router-dom";
import { HeroScreen } from "../../components/heroes/HeroScreen";

const { mount } = require("enzyme");

describe("Pruebas sobre HeroScreen", () => {
    const historyMock = {
      length: 10,
      goBack: jest.fn(),
      push: jest.fn(),
    };
 

  test("Debe de mostrar el componente Redirect si no hay argumentos en el url", () => {

    const wrapper = mount(
      <MemoryRouter initialEntries={["/hero"]}>
        <HeroScreen history={historyMock} />
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("Redirect").exists()).toBe(true);
  });

  test("Debe renderizar el componente heroScreen", () => {
      
    const wrapper = mount(
        <MemoryRouter initialEntries={["/hero/marvel-spider"]}>
            <Route path='/hero/:heroeId' component={HeroScreen}/>
        </MemoryRouter>
      );

        expect( wrapper ).toMatchSnapshot();

        expect(wrapper.find('.row').exists()).toBe(true);

  });

  test('debe de regresar a la pantalla anterior con PUSH', () => {
    
     const historyMock = {
       length: 1,
       goBack: jest.fn(),
       push: jest.fn(),
     };

     const wrapper = mount(
       <MemoryRouter initialEntries={["/hero/marvel-spider"]}>
         <Route 
            path="/hero/:heroeId" 
            component={ () => <HeroScreen history={historyMock}/> } 
        />
       </MemoryRouter>
     );

     wrapper.find('button').prop('onClick')();

     expect( historyMock.push ).toHaveBeenCalledWith('/')
     expect( historyMock.goBack ).not.toHaveBeenCalled();


  })

  test('History mayor a 2 goBack se haya llamado y no el push', () => {
    
     

     const wrapper = mount(
       <MemoryRouter initialEntries={["/hero/marvel-spider"]}>
         <Route
           path="/hero/:heroeId"
           component={() => <HeroScreen history={historyMock} />}
         />
       </MemoryRouter>
     );

     wrapper.find("button").prop("onClick")();

     expect(historyMock.goBack).toHaveBeenCalled();
     expect(historyMock.push).not.toHaveBeenCalled();

    
  });
  
  test('Debe de llamar el Redirect si el heroe no existe', () => {
      
    const wrapper = mount(
      <MemoryRouter initialEntries={["/hero/marvel-spider123123"]}>
        <Route
          path="/hero/:heroeId"
          component={() => <HeroScreen history={historyMock} />}
        />
      </MemoryRouter>
    );

    expect( wrapper.text() ).toBe('') 

  })
  
  
});
