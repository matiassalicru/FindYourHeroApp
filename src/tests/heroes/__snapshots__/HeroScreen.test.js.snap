// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre HeroScreen Debe de mostrar el componente Redirect si no hay argumentos en el url 1`] = `""`;

exports[`Pruebas sobre HeroScreen Debe renderizar el componente heroScreen 1`] = `
<div
  className="row mt-5"
>
  <div
    className="col-4"
  >
    <img
      alt="Spider Man"
      className="img-thumbnail animate__animated animate__fadeInLeft"
      src="../assets/heroes/marvel-spider.jpg"
    />
  </div>
  <div
    className="col8 animate__animated animate__fadeInLeft"
  >
    <h3>
      Spider Man
    </h3>
    <ul
      className="list-group list-group-flush "
    >
      <li
        className="list-group-item"
      >
        <b>
           Alter ego:
        </b>
         
        Peter Parker
      </li>
      <li
        className="list-group-item"
      >
        <b>
          Publisher: 
        </b>
         
        Marvel Comics
      </li>
      <li
        className="list-group-item"
      >
        <b>
          Appearance: 
        </b>
         
        Amazing Fantasy #15
      </li>
    </ul>
    <h5>
       characters 
    </h5>
    <p>
      Peter Parker
    </p>
    <button
      className="btn btn-outline-info"
      onClick={[Function]}
    >
      Return
    </button>
  </div>
</div>
`;
